torchao/quantization/quantize_/workflows/float8/float8_tensor.py

```diff
@@ -334,28 +373,25 @@ def _(func, types, args, kwargs):
             assert kernel_choice == "torch"
             scaled_mm_config = weight_tensor.mm_config
             assert scaled_mm_config is not None
-            out_shape = get_out_shape(input_tensor.shape, weight_tensor.shape)
+            out_shape = (*input_tensor.shape[:-1], weight_tensor.shape[1])
```

Comment:
technically `addmm` and `mm` don't support broadcasting of the gemm operands, would be more correct to enforce `x` and `w` to be 2d here and move the 3d shape support to `matmul` and `linear`.  If you want to do a TODO to punt, sounds ok to me.